lib/attack.go

```diff
@@ -223,7 +223,61 @@ func Client(c *http.Client) func(*Attacker) {
 	return func(a *Attacker) { a.client = *c }
 }
 
-// A Rate of hits during an Attack.
+// A Rater defines the rate of hits during an Attack by
+// returning the duration an Attacker should wait until hiting
+// the next Target. When a negative value time is returned,
+// an Attacker will stop.
+type Rater interface {
+	Wait(began, now time.Time) time.Duration
+}
+
+// A RaterFunc is a function adapter type that implements the
+// Rater interface.
+type RaterFunc func(began, now time.Time) time.Duration
+
+// Wait implements the Rater interface.
+func (f RaterFunc) Wait(began, now time.Time) time.Duration {
+	return f(began, now)
+}
+
+// A FixedRater is a Rater that produces hits at a fixed rate
+// for a given duration. It is not safe for concurrent use.
```

Comment:
It could easily be made concurrent safe, but I couldn't see use cases that'd justify that need. Do you think it'd be useful, for instance, to use the same `FixedRater` in two concurrent attacks in separate `Attackers`?